src/zebras/_freq.py

```diff
@@ -126,21 +136,39 @@ def __get_validators__(cls):
         yield freq
 
     @classmethod
-    def from_pandas(cls, freq) -> Freq:
-        if not isinstance(freq, str):
-            if hasattr(freq, "freqstr"):
-                freq = freq.freqstr
-            else:
-                raise ValueError(f"Invalid freq {freq}: {type(freq)}")
+    def get_weekday_offset(cls, freqstr):
+        if freqstr.upper() == "W":
+            return 0
+
+        assert (
+            len(freqstr) == 5
+            and freqstr[:2] == "W-"
+            and freqstr[2:] in list(day_offsets.keys())
+        ), "Supported options of freqstr are W-SUN, W-MON, W-TUE, W-WED, W-THU, W-FRI and W-SAT."
+
+        day = freqstr.split("-")[1]
+        return day_offsets[day]
 
+    @classmethod
+    def freq_name_n(cls, freq: str) -> [str, int]:
         match = maybe.expect(
             re.match(r"(?P<n>\d+)?\s*(?P<freq>(?:\w|\-)+)", freq),
             f"Unsupported freq format {freq}",
         )
         groups = match.groupdict()
 
-        name = groups["freq"].upper().split("-")[0]
-        n = maybe.map_or(groups["n"], int, 1)
+        return groups["freq"], maybe.map_or(groups["n"], int, 1)
+
+    @classmethod
+    def from_pandas(cls, freq) -> Freq:
+        if not isinstance(freq, str):
+            if hasattr(freq, "freqstr"):
+                freq = freq.freqstr
+            else:
+                raise ValueError(f"Invalid freq {freq}: {type(freq)}")
+
+        name, n = cls.freq_name_n(freq)
+        name = name.upper().split("-")[0]
```

Comment:
Maybe we should add the logic for `W-XXX` here?

Reply:
I think we don't want to store `W-XXX` in `Freq.name` since it is not recognized by `np`. I thought of keeping this function intact since we want `np` frequencies and just need the day offset to modify `Period.data`.

Reply:
We don't use frequencies of `numpy` but those defined by `pandas`, see:

https://github.com/awslabs/gluonts/blob/04919a894a4750c64031656689ff58b98df8757a/src/zebras/_freq.py#L85-L89

Reply:
Yes, you're right. Actually, I was referring to `numpy.datetime64` objects which are used internally for eg. in period calculation. But what exactly do you propose to do here? I didn't get it.